samza-sql/src/main/java/org/apache/samza/sql/translator/QueryTranslator.java

```diff
@@ -65,6 +66,7 @@ public void translate(SamzaSqlQueryParser.QueryInfo queryInfo, StreamGraph strea
     final TranslatorContext context = new TranslatorContext(streamGraph, relRoot, executionContext);
     final RelNode node = relRoot.project();
     final int[] joinId = new int[1];
+    final int[] windowId = new int[1];
```

Comment:
Looks like both windowId and joinId is used only within in the anonymous fn. I think you can move them  inside the anonymous class RelShuttleImpl, that ways you could convert them into a regular integer rather than a single valued array?